test/test_build_python.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2019 Rover Robotics
+# Licensed under the Apache License, Version 2.0
+import asyncio
+from pathlib import Path
+from types import SimpleNamespace
+
+import colcon_core.shell
+from colcon_core.package_descriptor import PackageDescriptor
+from colcon_core.plugin_system import SkipExtensionException
+from colcon_core.shell.bat import BatShell
+from colcon_core.shell.sh import ShShell
+from colcon_core.subprocess import new_event_loop
+from colcon_core.task import TaskContext
+from colcon_core.task.python.build import PythonBuildTask
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def monkey_patch_get_shell_extensions(monkeypatch):
```

Comment:
I would suggest to refactor this function to:

```
for shell in (ShShell, BatShell):
    try:
        instance = shell()
    except SkipExtensionException:
        continue
    monkeypatch.setattr(...)
    return
assert False, 'Failed to select primary shell extension'
```

Reply:
I'd rather keep it as is. It more clearly expresses that, after the loop, we have identified a single shell valid for the platform. Also, moving the monkeypatch out of that statement shows more clearly what we're patching and how.